src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -788,6 +788,24 @@ private bool IsValidExtensionMethodSignature()
                 this.ParameterRefKinds.IsDefault && // No 'ref' or 'out'
                 !this.IsParams();
 
+        private bool IsValidUserDefinedOperatorIs()
+        {
+            foreach (var parameter in this.Parameters)
+            {
+                if (parameter.RefKind != ((parameter.Ordinal == 0) ? RefKind.None : RefKind.Out))
+                {
+                    return false;
+                }
+            }
+
+            return
+                (this.ReturnsVoid || this.ReturnType.SpecialType != SpecialType.System_Boolean) &&
+                !this.IsGenericMethod &&
+                !this.IsVararg &&
+                this.ParameterCount > 0 &&
```

Comment:
There is some degree of code duplication between this method and SourceUserDefinedOperatorSymbolBase.CheckIsSignature. At the same time, there are some inconsistencies. For example, around nullable types. Consider merging the logic into a single method that can be used from both symbols. Diagnostics reporting could be conditioned on whether a non-null bag is passed in.